### ISSUE TITLE
NTCurve Needs a way to create splines from INTERP pts as well as control Pts

### ISSUE THREAD

**Issue by matthewleta** [nTopology/juce_icarus#7]
for surface mapping, i need all points to be inside the bounds of the template. with splines, however, those are generated from control points. even if the spline itself is inside the bounds of the template, a control point can easily be outside, and i wont be able to map it. so we need a way to get points from the line, as well as make an interpolated line from the same point set.

 any thoughts?

**Comment by bradrothenberg** [comment]
yeah -- can you use the points from the segmentSpline function ? rather than map splines directly, do you want to just map polylines that have a lot of points?

I am going to run into this same problem -- I was just assuming the control points for the spline would always be inside the cube for the units, but I guess it makes sense that they can be outside?

Do we really want to get into figuring out how to draw interpolated curves? maybe cgal has this?

**Comment by matthewleta** [comment]
well if we say that all curves are actually polylines then we dont really have to. but an interpolated curve point set will be virtually the same as a polyline point set

**Comment by bradrothenberg** [comment]
maybe we use this? http://doc.cgal.org/latest/Polyline_simplification_2/index.html#Chapter_2D_Polyline_simplification

**Comment by matthewleta** [comment]
this will be good for minimizing redundant points in a polyline, so if we are going to do polylines, then this would be great

**Comment by matthewleta** [comment]
if we just did uniform segmentation, then there would be unecessary points if the curve had straight areas

**Comment by bradrothenberg** [comment]
yup, but that module only works on 2d lines

**Comment by matthewleta** [comment]
oh. then thats no good

**Comment by matthewleta** [comment]
that might be easy to write though

**Comment by bradrothenberg** [comment]
Can you just assume the points will be in the unit, or at least in the neighbor units & you can do the neighbor unit mapping you had from before?

**Comment by matthewleta** [comment]
yeah totally. it will be the same function regardless. what i brought up was more of a "in the future" thing
